themes/default/content/docs/intro/concepts/inputs-outputs.md

```diff
@@ -651,6 +651,327 @@ variables:
+
+```go
+jsonIAMPolicy := pulumi.ToOutput(`{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}`)
+
+policyWithNoStatements := pulumi.JSONUnmarshal(jsonIAMPolicy.(pulumi.StringInput)).ApplyT(func(v interface{}) (interface{}, error) {
+    // delete the policy statements
+    v.(map[string]interface{})["Statement"] = []pulumi.ArrayOutput{}
+    return v, nil
+})
+```
+
+For more details view the Go documentation.
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+### Example Converting Outputs to JSON
+
+The following example allows you to natively represent the definition of an AWS Step Function State Machine and easily embed outputs from other resources in your stack.
+
+```csharp
+var stateMachine = Pulumi.Output.JsonSerialize(Output.Create(new {
+        Comment = "A Hello World example of the Amazon States Language using two AWS Lambda Functions",
+        StartAt = "Hello",
+        States = new Dictionary<string, object?>{
+        ["Hello"] = new {
+            Type = "Task",
+            Resource = helloFunction.Arn,
+            Next = "World",
+        },
+        ["World"] = new {
+            Type = "Task",
+            Resource = worldFunction.Arn,
+            End = true,
+        },
+    },
+}));
+```
+
+### Example Parsing JSON Outputs to Objects
+
+The following is an example of Pulumi code that operates on a JSON object and removes all of the policy statements by replacing it with an empty list.
+
+```csharp
+var jsonIAMPolicy = Output.Create(@"
+        {
+            ""Version"": ""2012-10-17"",
+            ""Statement"": [
+                {
+                    ""Sid"": ""VisualEditor0"",
+                    ""Effect"": ""Allow"",
+                    ""Action"": [
+                        ""s3:ListAllMyBuckets"",
+                        ""s3:GetBucketLocation""
+                    ],
+                    ""Resource"": ""*""
+                },
+                {
+                    ""Sid"": ""VisualEditor1"",
+                    ""Effect"": ""Allow"",
+                    ""Action"": [
+                        ""s3:*""
+                    ],
+                    ""Resource"": ""arn:aws:s3:::my-bucket""
+                }
+            ]
+        }
+    ");
+
+var policyWithNoStatements = Pulumi.Output.JsonDeserialize<IAMPolicy>(jsonIAMPolicy).Apply(policy =>
+{
+    // delete the policy statements.
+    policy.Statement = Pulumi.Output.Create(new List<StatementEntry> { });
+    return policy;
+})
+```
+
+For more details view the .NET documentation.
+
+{{% /choosable %}}
```

Comment:
@Zaid-Ajaj @Frassle does this look idiomatic?

Reply:
Looks ok to me.